code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm

```diff
@@ -10,153 +12,140 @@
-					"<span class='notice'>You hear squelching.</span>")
-				resisting = 1
-				var/mob/oldbuckled = buckled_mob
-				spawn(nest_resist_time)
-					if(resisting && buckled_mob == oldbuckled && buckled_mob.stat != DEAD) //Must be alive and conscious
-						resisting = 0
-						resisting_ready = 1
-						if(ishuman(usr))
-							var/mob/living/carbon/human/H = usr
-							if(H.handcuffed)
-								to_chat(buckled_mob, "<span class='danger'>You are ready to break free of the nest, but your limbs are still secured. Resist once more to pop up, then resist again to break your limbs free!</span>")
-							else
-								to_chat(buckled_mob, "<span class='danger'>You are ready to break free! Resist once more to free yourself!</span>")
-			add_fingerprint(user)
-
-/mob/living/carbon/human/proc/start_nesting_cooldown()
-	set waitfor = 0
-	recently_unbuckled = 1
-	sleep(50)
-	recently_unbuckled = 0
-
-/obj/structure/bed/nest/buckle_mob(mob/M as mob, mob/user as mob)
-
-	if(!ismob(M) || (get_dist(src, user) > 1) || (M.loc != loc) || user.restrained() || user.stat || user.lying || M.buckled || !iscarbon(user))
-		return
 
-	if(buckled_mob)
-		to_chat(user, "<span class='warning'>There's already someone in [src].</span>")
+/obj/structure/bed/nest/manual_unbuckle(mob/living/user)
+	if(!buckled_mob || buckled_mob.buckled != src)
 		return
 
-	if(M.mob_size > MOB_SIZE_HUMAN)
-		to_chat(user, "<span class='warning'>\The [M] is too big to fit in [src].</span>")
-		return
+	if(buckled_mob != user)
+		if(user.incapacitated())
+			return
+		buckled_mob.visible_message("<span class='notice'>\The [user] pulls \the [buckled_mob] free from \the [src]!</span>",\
+		"<span class='notice'>\The [user] pulls you free from \the [src].</span>",\
+		"<span class='notice'>You hear squelching.</span>")
+		playsound(loc, "alien_resin_move", 50)
+		if(ishuman(buckled_mob))
+			var/mob/living/carbon/human/H = buckled_mob
+			H.last_unbuckled = world.time
+		unbuckle()
+	else
+		if(buckled_mob.incapacitated(TRUE))
+			to_chat(buckled_mob, "<span class='warning'>You're currently unable to try that.</span>")
+			return
+		if(!resisting_time)
+			resisting_time = world.time
+			buckled_mob.visible_message("<span class='warning'>\The [buckled_mob] struggles to break free of \the [src].</span>",\
+			"<span class='warning'>You struggle to break free from \the [src].</span>",\
+			"<span class='notice'>You hear squelching.</span>")
+			addtimer(CALLBACK(src, .proc/unbuckle_time_message, user), NEST_RESIST_TIME)
+			return
+		if(resisting_time + NEST_RESIST_TIME > world.time)
+			to_chat(buckled_mob, "<span class='warning'>You're already trying to free yourself. Give it some time.</span>")
+			return
+		buckled_mob.visible_message("<span class='danger'>\The [buckled_mob] breaks free from \the [src]!</span>",\
+		"<span class='danger'>You pull yourself free from \the [src]!</span>",\
+		"<span class='notice'>You hear squelching.</span>")
+		unbuckle()
+
+	add_fingerprint(user)
```

Comment:
put this at the start

Reply:
That will generate some quirky behavior, but sure.